### PR TITLE
Adding loader for freefield1010

### PR DESCRIPTION
**Pull request by tanmayy24**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`.
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`.
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`.
- [x] Run `black`, `flake8` and `mypy` (see [Running your tests locally](https://soundata.github.io/en/latest/source/contributing.html#running-your-tests-locally)).
- [x] Run `tests/test_full_dataset.py` on your dataset.
- [x] Check that codecov coverage does not decrease.

### CODE REVIEW

File: docs/source/soundata.rst [hunk 1]
```diff
@@ -62,6 +62,13 @@ ESC-50
    :members:
    :inherited-members:
 
+freefield1010
+^^^^^^
```
Comment (guillemcortes) [inline docs/source/soundata.rst]: extend `^^^^^^` it matches freefield1010 length

File: docs/source/table.rst [hunk 2]
```diff
@@ -74,6 +74,15 @@
      - .. image:: https://licensebuttons.net/l/by-nc/3.0/80x15.png
           :target: https://creativecommons.org/licenses/by-nc/3.0
 
+   * - freefield1010
```
Comment (guillemcortes) [inline docs/source/table.rst]: is `freefield1010` the official name of the dataset? Does it have a capitalized version?

File: scripts/make_freefield1010_index.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+def main(args):
+    if os.path.exists(args.data_path):
+        make_index(args.data_path)
+    else:
+        logging.error("Invalid data_path: %s does not exist.", args.data_path)
+    
+
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser(description="Generate freefield1010 index file.")
```
Comment (guillemcortes) [inline scripts/make_freefield1010_index.py]: Change `PARSER` to `parser`

File: soundata/datasets/freefield1010.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
+"""freefield1010 Dataset Loader
+
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    Created By
+    ----------
```
Comment (guillemcortes) [inline soundata/datasets/freefield1010.py]: Maybe we could use the following section names formatting instead? (I see that is the most common):

```*FSD50K: an Open Dataset of Human-Labeled Sound Events*

    *Created By:*
        | Eduardo Fonseca, Xavier Favory, Jordi Pons, Frederic Font, Xavier Serra.
        | Music Technology Group, Universitat Pompeu Fabra (Barcelona). 
        
    Version 1.0

    *Description:*
        FSD50K is an open dataset of human-labeled sound events containing 51,197 Freesound clips unequally
        distributed in 200 classes drawn from the AudioSet Ontology. FSD50K has been created at the Music Technology Group
        of Universitat Pompeu Fabra.

    *Audio Files Included:*
        * FSD50K contains 51,197 audio.......```

**Review by guillemcortes — CHANGES_REQUESTED** [review]
Super minor changes. The rest looks good to me.

File: docs/source/table.rst (hunk 2 above)
Comment (tanmayy24) [reply docs/source/table.rst:77]: Yes, I see freefield1010 being used as the official name for the dataset

**Review by guillemcortes — APPROVED** [review]
Good, ready to merge!

**Pull request merged**